### PR TITLE
ENH: Adding ComposeScaleSkewVersor 3D Transform

### PR DESCRIPTION
**Pull request by aylward**
This transform uses matrix compose to combine matrix defined by
versor, scale, and skew parameters.  Implementing the
jacobian of this transform is left as an exercise for future
developers.

### CODE REVIEW

**Comment by dzenanz** [comment]
The new class is not Python-wrapped.

**Comment by aylward** [comment]
Added python wrapping.

File: Modules/Core/Transform/include/itkComposeScaleSkewVersor3DTransform.hxx [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+
+  jacobian[0][2] = 2.0 * (-2 * vzw * px + (vzz - vww) * py + (vxw - vyz) * pz) / vw;
+  jacobian[1][2] = 2.0 * ((vww - vzz) * px - 2 * vzw * py + (vyw + vxz) * pz) / vw;
+  jacobian[2][2] = 2.0 * ((vxw + vyz) * px + (vyw - vxz) * py) / vw;
+
+  jacobian[0][3] = 1.0;
+  jacobian[1][4] = 1.0;
+  jacobian[2][5] = 1.0;
+
+  jacobian[0][6] = px;
```
Comment (blowekamp) [inline Modules/Core/Transform/include/itkComposeScaleSkewVersor3DTransform.hxx]: The scale dependent Jacobian values an incorrect as well.

Maybe this whole implementation should be removed and only leave the exception?

File: Modules/Core/Transform/test/itkComposeScaleSkewVersor3DTransformTest.cxx [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+/**
+ *
+ *  This program illustrates the use of Similarity3DTransform
```
Comment (blowekamp) [inline Modules/Core/Transform/test/itkComposeScaleSkewVersor3DTransformTest.cxx]: Similarity3DTransform->ComposeScaleSkewVersor3DTransform

File: Modules/Core/Transform/include/itkComposeScaleSkewVersor3DTransform.h [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+  itkGetConstReferenceMacro(Skew, SkewVectorType);
+
+  void
+  SetIdentity() override;
+
+  /** This method computes the Jacobian matrix of the transformation.
+   * given point or vector, returning the transformed point or
+   * vector. The rank of the Jacobian will also indicate if the
+   * transform is invertible at this point. */
```
Comment (blowekamp) [inline Modules/Core/Transform/include/itkComposeScaleSkewVersor3DTransform.h]: Change the comment to indicate that this is not implemented and an exception will be thrown.

File: Modules/Core/Transform/test/itkComposeScaleSkewVersor3DTransformTest.cxx (hunk 2 above)
Comment (dzenanz) [reply Modules/Core/Transform/test/itkComposeScaleSkewVersor3DTransformTest.cxx]: The same change should probably be applied to the original class too?

File: Modules/Core/Transform/include/itkComposeScaleSkewVersor3DTransform.hxx (hunk 1 above)
Comment (aylward) [reply Modules/Core/Transform/include/itkComposeScaleSkewVersor3DTransform.hxx]: Removed and appropriate comments added.

File: Modules/Core/Transform/test/itkComposeScaleSkewVersor3DTransformTest.cxx (hunk 2 above)
Comment (aylward) [reply Modules/Core/Transform/test/itkComposeScaleSkewVersor3DTransformTest.cxx]: Changed in new and in old/original class too.

File: Modules/Core/Transform/include/itkComposeScaleSkewVersor3DTransform.h (hunk 3 above)
Comment (aylward) [reply Modules/Core/Transform/include/itkComposeScaleSkewVersor3DTransform.h]: Done.

**Pull request merged**